pkg/api/properties.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/pkg/model"
+)
+
+const (
+	// Minimal property set required to translate between GRPC API and model.
+	apiTypePackage              = "olm.package"
+	propertyTypeProvidedPackage = "olm.package.provided"
+	propertyTypeRequiredPackage = "olm.package.required"
+	apiTypeGVK                  = "olm.gvk"
+	propertyTypeProvidedGVK     = "olm.gvk.provided"
+	propertyTypeRequiredGVK     = "olm.gvk.required"
+	propertyTypeChannel         = "olm.channel"
+	propertyTypeSkips           = "olm.skips"
+	propertyTypeSkipRange       = "olm.skipRange"
+)
+
+type channel struct {
+	Name     string `json:"name"`
+	Replaces string `json:"replaces,omitempty"`
+}
+
+type providedPackage struct {
+	PackageName string `json:"packageName"`
+	Version     string `json:"version"`
+}
+
+type properties struct {
+	providedPackage providedPackage
+	providedGVKs    []*GroupVersionKind
+	requiredGVKs    []*GroupVersionKind
+	skipRange       string
+}
+
+func parseProperties(props []model.Property) (*properties, error) {
```

Comment:
Some of this code is more or less duplicated in the `pkg/api` package. It initially seemed like a good idea to scope the property parsing to the component that needs it and then ONLY parse the properties required by the component.

But some properties have two or more components that are interested in them, so it seems reasonable to centralize property definitions and parsing code as well. Thoughts?